AdobeStockImage/Model/Storage.php

```diff
@@ -71,24 +72,25 @@ public function __construct(
     /**
      * Save file from the URL to destination directory relative to media directory
      *
-     * @param string $imageUrl
+     * @param Asset $assets
      * @param string $destinationDirectoryPath
      * @return string
      * @throws LocalizedException
      */
-    public function save(string $imageUrl, string $destinationDirectoryPath = '') : string
+    public function save(Asset $assets, string $destinationDirectoryPath = '') : string
     {
         if (!empty($destinationDirectoryPath)) {
             $destinationDirectoryPath = rtrim($destinationDirectoryPath, '/') . '/';
         }
-        $destinationPath = $destinationDirectoryPath . $this->getFileName($imageUrl);
+        $imgName = $this->generateImageName($assets->getData());
```

Comment:
Please avoid unnecessary variables shortening, better readability is a priority in comparison to saving 2 chars
```suggestion
        $imageName = $this->generateImageName($assets->getData());
```